src/inference.jl

```diff
@@ -326,7 +332,8 @@ function sample(
                 # apply softmax to the logits to get the probabilities for next token
                 softmax!(state.logits)
                 # sample from this distribution to get the next token
-                next = wsample(1:config.vocab_size, state.logits)
+                logits = Array(state.logits) # copy to CPU since wsample is not supported on GPU
```

Comment:
Interesting -- probably a good place to pick up some speed. It looks like someone wrote about this [here](https://tianjun.me/essays/Categorical_Sampling_on_GPU_with_Julia/) but it's probably not important for the moment.

Reply:
Yeah, worth noting that one, because the `logits` is a 128000+ length vector.

Reply:
Let's flag it for later and do a separate PR on it. Do you think there's a good upstream place we could put this? Like a kernel library somewhere.